### PR TITLE
switch: have savefiles specific to each user

### PR DESCRIPTION
**Pull request by terabyte25**
Like skyrim.

(I haven't even tested out or compiled these changes, so they are probably horribly unsafe and won't compile)

### CODE REVIEW

**Comment by fgsfdsfgs** [comment]
Maybe move `accountInitialize` and `accountExit` to `userAppInit`/`userAppExit`? The rest should probably be done in `Switch::startup` or something in that case. Or just move the entire thing to `startup`.
Additionally, you're assigning C strings (which would work if the variable was a `char *`).

**Comment by terabyte25** [comment]
Yeah, I figure I'll just move the whole thing with usernames to startup.

For the c strings, I guess I'll just make username a c++ string then to assign the profile username I'll just do 
```
username = std::string(profilebase.username);
```
 (I'm not very proficient with how c does strings, I just use c++'s strings most of the time)

**Comment by terabyte25** [comment]
Should be good now.

File: apps/openmw/switch_startup.cpp [hunk 1]
```diff
@@ -130,6 +130,33 @@ void Switch::startup()
     appletLockExit();
 }
 
+void Switch::getUsername() {
```
Comment (fgsfdsfgs) [inline apps/openmw/switch_startup.cpp]: `std::string Switch::getUsername()`

Comment (terabyte25) [reply apps/openmw/switch_startup.cpp]: Damn it. I keep doing that for some reason

File: apps/openmw/switch_startup.cpp [hunk 2]
```diff
@@ -130,6 +130,33 @@ void Switch::startup()
     appletLockExit();
 }
 
+void Switch::getUsername() {
+    // get the name of the current user
+    accountIntialize();
```
Comment (fgsfdsfgs) [inline apps/openmw/switch_startup.cpp]: accountIn**i**tialize

**Comment by fgsfdsfgs** [comment]
Builds now.

**Comment by terabyte25** [comment]
Probably doesn't really work because I didn't do the RC checks (like in the example) and what not.

**Comment by fgsfdsfgs** [comment]
It works, although I'm going to change it a little bit.

**Comment by fgsfdsfgs** [comment]
Now that I think about it, it could be a bad idea because usernames can contain UTF-8 and I have no idea if that's handled correctly in the fs driver.

**Comment by terabyte25** [comment]
I think boost has built in support for UTF8?

**Comment by fgsfdsfgs** [comment]
Boost might, however `fs` and other things might not.
I have checked, names containing Unicode characters make it silently die when it tries to create the folder.
FAT32 also doesn't support UTF-8 directly, IIRC Microsoft uses some wacky workarounds for that, but we have a different fs driver.

**Comment by terabyte25** [comment]
http://doc.aldebaran.com/1-14/ref/libqi/boost_filesystem_example_8cpp-example.html

**Comment by terabyte25** [comment]
Does libnx's functions return it in utf8?

**Comment by fgsfdsfgs** [comment]
Yes. I have obtained the error message it crashes with and it seems to indicate that `mkdir` or the underlying `fs` call cannot create a path that has unicode in it, because it attempts to create the directory and right after that I get
```
Error: boost::filesystem::create_directory: No such file or directory: "./data/いしにさ"
```

**Comment by fgsfdsfgs** [comment]
It's actually possible to create a folder with Cyrillic in the name, but not Japanese. Wonder why.
UPD: Thought that this had to do with system locale or something, but changing the region to Japanese doesn't fix it. I think I'll revert it for the time being, if you have no other suggestions.

**Comment by terabyte25** [comment]
For now I think a band-aid fix would be to detect if the username is non ASCII, and if it is just use the fall back.

**Comment by terabyte25** [comment]
Maybe you might get a result with messing around with the locales?

```
std::local loc=boost::locale::generator().generate(""); 

std::locale::global(loc);

boost::filesystem::path::imbue(std::locale()); 

```

**Comment by fgsfdsfgs** [comment]
Nah, it seems to be a bug with fs. It just doesn't deal with certain character ranges apparently.

**Pull request merged**